components/did/Core/Src/DID.c

```diff
@@ -250,22 +252,53 @@ void update_DID_screen()
 				OutputString("---", STATE_DATA_XPOS, STATE_DATA_YPOS);
 			}
 
-			// Target cruise
-			if ( target_velocity != NULL && MCB_drive_state != NULL )
+			// Regen Enable
+			if ( MCB_regen_enabled != NULL )
 			{
-				if ( *MCB_drive_state == 0x02 )
-				{
-					UpdateScreenParameter(CRUISE_DATA_XPOS, CRUISE_DATA_YPOS, (uint32_t)(*target_velocity), 0, FALSE);
-				}
-				else
-				{
-					OutputString("OFF", CRUISE_DATA_XPOS, CRUISE_DATA_YPOS);
+				if (*MCB_regen_enabled == ENABLED) {
+					OutputString("   ", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
+					OutputString("ON", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
+				} else {
+					OutputString("   ", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
+					OutputString("OFF", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
 				}
 			}
 			else
 			{
-				OutputString("---", CRUISE_DATA_XPOS, CRUISE_DATA_YPOS);
+				OutputString("---", REGEN_DATA_XPOS, REGEN_DATA_YPOS);
 			}
+
+			// Pack Voltage
+			if( pack_voltage != NULL )
+			{
+				UpdateScreenParameter(PACK_VOLT_DATA_XPOS, PACK_VOLT_DATA_YPOS, (uint32_t)(*pack_voltage), 0, FALSE);
+			}
+			else
+			{
+				OutputString("---", PACK_VOLT_DATA_XPOS, PACK_VOLT_DATA_YPOS);
+			}
+
+
+//			// Simulation target speed
+//			if( simulation_speed != NULL )
+//			{
+//				UpdateScreenParameter(TARGET_DATA_XPOS, TARGET_DATA_YPOS, (uint32_t)(*simulation_speed), 0, FALSE);
+//			}
+//			else
+//			{
+//				OutputString("---", TARGET_DATA_XPOS, TARGET_DATA_YPOS);
+//			}
+
+			// Pack Current
+			if ( pack_current != NULL )
+			{
+				UpdateScreenParameter(PACK_CURRENT_DATA_XPOS, PACK_CURRENT_DATA_YPOS, (uint32_t)(*pack_current), ((uint8_t)((*pack_current) * 10) % 10), TRUE);
```

Comment:
Why was the (packcurrent * 10) % 10 calculation chosen. Can we add a comment here for it.

Reply:
That gives us the decimal point.